### PR TITLE
Add a new keyboard: yoichiro/lunakey_macro

### PR DESCRIPTION
**Pull request by yoichiro**
This pull request adds a new keyboard designed by @yoichiro to the QMK Firmware.

## Description

I sent this pull request to add a new keyboard named "Lunakey Macro" to the QMK Firmware. Because this keyboard was designed by @yoichiro and this keyboard belongs to the Lunakey family, I want to add this keyboard definition into the `yoichiro` directory.

And, I created the simple default key map definition and the key map definition for VIA support.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yoichiro/lunakey_macro/config.h [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+/*== or choose animations ==*/
+//#    define RGBLIGHT_EFFECT_BREATHING
+//  #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//#define RGBLIGHT_EFFECT_SNAKE
+//#    define RGBLIGHT_EFFECT_KNIGHT
+//#define RGBLIGHT_EFFECT_CHRISTMAS
+//#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//#define RGBLIGHT_EFFECT_RGB_TEST
+//#    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/yoichiro/lunakey_macro/config.h:92]: could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/yoichiro/lunakey_macro/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1,3 @@
+AUDIO_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/yoichiro/lunakey_macro/keymaps/default/rules.mk:3]: These shouldn't be enabled at the keymap level, for the default keymap.  Instead, they should be at the keyboard level.

File: keyboards/yoichiro/lunakey_macro/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+![Lunakey Macro](imgur.com image replace me!)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [Yoichiro Tanaka](https://github.com/yourusername)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/yoichiro/lunakey_macro/readme.md:11]: Could you add instructions on hw to enter the bootloader?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yoichiro/lunakey_macro/keymaps/default/rules.mk (hunk 2 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_macro/keymaps/default/rules.mk:3]: @drashna Thanks, I remove the flags from the default keymap definition.

File: keyboards/yoichiro/lunakey_macro/readme.md (hunk 3 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_macro/readme.md:11]: @drashna Copy that.

**Comment by yoichiro** [comment]
@drashna I modified some lines according to your suggestions at 3ae4050. Could you review them again? Thanks!

**Review by drashna — APPROVED** [review]

File: keyboards/yoichiro/lunakey_macro/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+* Hardware Supported: Lunakey Macro PCB, Pro Micro MCU.
+* Hardware Availability: Published yet.
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lunakey_macro:default
+
+Flashing example for this keyboard:
+
+    make lunakey_macro:default:avrdude
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_macro/readme.md:17]:
```suggestion
    make lunakey_macro:default:flash
```

File: keyboards/yoichiro/lunakey_macro/readme.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Lunakey Macro
+
+![Lunakey Macro](https://i.imgur.com/BirFxgY.jpeg)
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_macro/readme.md:3]:
```suggestion
![Lunakey Macro](https://i.imgur.com/BirFxgYh.jpeg)
```

File: keyboards/yoichiro/lunakey_macro/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    ),
+    [_CURSOR] = LAYOUT(
+        KC_PGUP,      KC_HOME,    KC_UP,       KC_END, \
+        KC_PGDN,      KC_LEFT,    KC_DOWN,     KC_RGHT, \
+        KC_ENT,       KC_BSPC,    DF(_ADJUST) \
+    ),
+    [_ADJUST] = LAYOUT(
+        RGB_TOG,      RGB_MOD,    RGB_VAI,     RGB_VAD, \
+        AU_TOG,       CK_TOGG,    MU_TOG,      MU_MOD, \
+        RESET,        KC_NO,      DF(_NUMBERS) \
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_macro/keymaps/via/keymap.c:39]:
```suggestion
        KC_1,         KC_2,       KC_3,        KC_4,
        KC_5,         KC_6,       KC_7,        KC_8,
        KC_9,         KC_0,       DF(_CURSOR)
    ),
    [_CURSOR] = LAYOUT(
        KC_PGUP,      KC_HOME,    KC_UP,       KC_END,
        KC_PGDN,      KC_LEFT,    KC_DOWN,     KC_RGHT,
        KC_ENT,       KC_BSPC,    DF(_ADJUST)
    ),
    [_ADJUST] = LAYOUT(
        RGB_TOG,      RGB_MOD,    RGB_VAI,     RGB_VAD,
        AU_TOG,       CK_TOGG,    MU_TOG,      MU_MOD,
        RESET,        KC_NO,      DF(_NUMBERS)
```

File: keyboards/yoichiro/lunakey_macro/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    ),
+    [_CURSOR] = LAYOUT(
+        KC_PGUP,      KC_HOME,    KC_UP,       KC_END, \
+        KC_PGDN,      KC_LEFT,    KC_DOWN,     KC_RGHT, \
+        KC_ENT,       KC_BSPC,    DF(_ADJUST) \
+    ),
+    [_ADJUST] = LAYOUT(
+        KC_NO,        KC_NO,      KC_NO,       KC_NO, \
+        KC_NO,        KC_NO,      KC_NO,       KC_NO, \
+        RESET,        KC_NO,      DF(_NUMBERS) \
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_macro/keymaps/default/keymap.c:39]:
```suggestion
        KC_1,         KC_2,       KC_3,        KC_4,
        KC_5,         KC_6,       KC_7,        KC_8,
        KC_9,         KC_0,       DF(_CURSOR)
    ),
    [_CURSOR] = LAYOUT(
        KC_PGUP,      KC_HOME,    KC_UP,       KC_END,
        KC_PGDN,      KC_LEFT,    KC_DOWN,     KC_RGHT,
        KC_ENT,       KC_BSPC,    DF(_ADJUST)
    ),
    [_ADJUST] = LAYOUT(
        KC_NO,        KC_NO,      KC_NO,       KC_NO,
        KC_NO,        KC_NO,      KC_NO,       KC_NO,
        RESET,        KC_NO,      DF(_NUMBERS)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by yoichiro** [comment]
@fauxpark Thank you for the suggestions. I committed all of them. Could you confirm them?

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by yoichiro** [comment]
@drashna @fauxpark Thank you for your reviewing!

**Comment by drashna** [comment]
> @drashna @fauxpark Thank you for your reviewing!

Well, Thank you for your contribution!

**Pull request merged**